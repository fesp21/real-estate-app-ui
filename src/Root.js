import React, { Component } from 'react';
import { StatusBar } from 'react-native';
import { Provider } from 'react-redux';
import { NavigationContext, NavigationProvider, StackNavigation } from '@exponent/ex-navigation';
import { NavigationStyles } from '@exponent/ex-navigation';
import Router from './common/router';
import Store from './common/store';
import CodePush from 'react-native-code-push';
import colors from './common/colors';
import { CODEPUSH_ENABLED } from './env';

const navigationContext = new NavigationContext({
  router: Router,
  store: Store,
});

export default class Root extends Component {

  constructor() {
    super();
    if (CODEPUSH_ENABLED) {
      CodePush.sync();
    }
  }

  componentWillMount() {
    StatusBar.setHidden(true);
  }

  componentWillUnmount() {
    StatusBar.setHidden(false);
  }

  render() {
    const modalStyle = {
      ...NavigationStyles.SlideVertical,
      gestures: null,
      // sceneAnimations: (props) => {
      //   const {
      //     position,
      //     scene,
      //   } = props;
      //
      //   const index = scene.index;
      //   const inputRange = [index - 1, index, index + 1];
      //
      //   const opacity = position.interpolate({
      //     inputRange,
      //     outputRange: [0, 1, 1],
      //   });
      //
      //   return {
      //     opacity,
      //     transform: [
      //       { translateX: 0 },
      //       { translateY: 0 },
      //       { scale: 1 },
      //     ],
      //     backgroundColor: 'transparent',
      //     shadowOpacity: 0,
      //   };
      // },
    };

    return (
      <Provider store={Store} >
        <NavigationProvider context={navigationContext}>
          <StackNavigation
            id="rootStack"
            navigatorUID="rootStack"
            initialRoute="app"
            defaultRouteConfig={{
              styles: { ...modalStyle },
              navigationBar: {
                backgroundColor: colors.lightGrey,
                tintColor: 'white',
              },
            }}
          />
        </NavigationProvider>
      </Provider>
    );
  }

}
