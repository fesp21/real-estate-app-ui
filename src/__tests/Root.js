// __tests__/Intro-test.js
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Root from '../Root';
import MockStorage from '../../__tests__/__mocks__/MockStorage';

const storageCache = {};
const AsyncStorage = new MockStorage(storageCache);

jest.setMock('AsyncStorage', AsyncStorage);

// Note: test renderer must be required after react-native.

test('renders Root Component', () => {
  const tree = renderer.create(
    <Root />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});