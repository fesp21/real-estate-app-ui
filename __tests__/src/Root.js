// __tests__/Intro-test.js
import 'react-native';
import React from 'react';
import Root from '../../src/Root';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders Root Component', () => {
  const tree = renderer.create(
    <Root />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});