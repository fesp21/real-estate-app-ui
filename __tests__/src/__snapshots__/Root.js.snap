exports[`test renders Root Component 1`] = `
<View
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        Object {
          "flex": 1,
        },
      ]
    }>
    <View
      style={
        Object {
          "flex": 1,
        }
      }>
      <View
        style={
          Object {
            "flex": 1,
          }
        }>
        <View
          needsOffscreenAlphaCompositing={true}
          pointerEvents="auto"
          style={
            Object {
              "backgroundColor": "white",
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0.4,
              "shadowRadius": 10,
              "top": 0,
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }>
          <View
            style={
              Object {
                "flex": 1,
              }
            }>
            <View
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }>
              <View
                style={
                  Object {
                    "flex": 1,
                  }
                } />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "left": 0,
              "position": "absolute",
              "right": 0,
            },
            Object {
              "top": 0,
            },
          ]
        }>
        <View />
      </View>
    </View>
  </View>
</View>
`;
